mono/profiler/proflog.h

```diff
@@ -16,6 +16,8 @@
  * version 8: added TYPE_RUNTIME and JIT helpers/trampolines
  * version 9: added MONO_PROFILER_CODE_BUFFER_EXCEPTION_HANDLING
  * version 10: added TYPE_COVERAGE
+ * version 11: added thread ID to TYPE_SAMPLE_HIT
+ * version 12: added more load/unload events
```

Comment:
I'm not sure the sorting counts as a format change. Any number of random factors can have an impact on ordering of events in a file; this is just another one, which should be completely transparent to consumers. They should just see better compression.

Regarding thread ID, as I mentioned on Slack, it was a trivial change to add a thread ID field to the `TYPE_SAMPLE_HIT` event, and it made sense to do since we're bumping the format version for all these new metadata events anyway. On the other hand, making the buffer header thread ID usable for these events was a lot of really ugly, hacky work because the profiler wasn't designed to handle the use case of a log buffer having a different thread ID than the actual thread writing to it.